dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/boogie/VisitorBoogie.java

```diff
@@ -96,13 +96,28 @@ public class VisitorBoogie extends BoogieBaseVisitor<Object> {
 	protected BeginAtomic currentBeginAtomic = null;
 	protected Call_cmdContext atomicMode = null;
 	 
-	private ExprSimplifier exprSimplifier = new ExprSimplifier();
-	
-	private final List<String> smackDummyVariables = Arrays.asList("$M.0", "$exn", "$exnv", "$CurrAddr", "$GLOBALS_BOTTOM", "$EXTERNS_BOTTOM", "$MALLOC_TOP", "__SMACK_code", "__SMACK_decls", "__SMACK_top_decl", "$1024.ref", "$0.ref", "$1.ref", ".str.1", "env_value_str", ".str.1.3", ".str.19", "errno_global", "$CurrAddr");
+	private final ExprSimplifier exprSimplifier = new ExprSimplifier();
+
 
 	public VisitorBoogie(ProgramBuilder pb) {
 		this.programBuilder = pb;
 	}
+
+	private final List<String> smackDummyVariables =
+			Arrays.asList("$M.0", "$exn", "$exnv", "$CurrAddr", "$GLOBALS_BOTTOM",
+			"$EXTERNS_BOTTOM", "$MALLOC_TOP", "__SMACK_code", "__SMACK_decls", "__SMACK_top_decl",
+			"$1024.ref", "$0.ref", "$1.ref", "env_value_str", "errno_global", "$CurrAddr");
+	private boolean doIgnoreVariable(String varName) {
+		// We ignore some smack-generated dummy variables
+		if (smackDummyVariables.contains(varName)) {
+			return true;
+		}
+		// We also ignore all kinds of strings for now.
+		if (varName.contains(".str")) {
```

Comment:
Maybe `varName.startsWith(".str")` to minimize changes of matching a real variable?

Reply:
We also skip user-declared string variables, no?
And name-wise, I don't think there are clashes because of the `.` in the front (I don't think real variables have a dot in them).

Reply:
I mean, if the user declares e.g., `int no.string`, we would skip this real variable.

Reply:
I don't think that this is a valid name. The only allowed special symbol is `_` I believe.